Update ROS launch documentation (frontend documents)

ivanpauno opened the pull request:
Address some comments in https://github.com/ros2/design/pull/163.

ivanpauno committed:
Address comments. Minimal update

Signed-off-by: ivanpauno <ivanpauno@ekumenlabs.com>

ivanpauno commented:
Also, added `<executable>` example and renamed after https://github.com/ros2/launch_ros/pull/47

hidmic commented on articles/151_roslaunch_xml.md:
```diff
@@ -114,6 +114,18 @@ Arguments are limited to the scope of their definition and thus have to be expli
+
+##### Description
+
+The `<executable>` tag allows for executing any executable in the form of local OS process.
```
@ivanpauno consider
```suggestion
The `<executable>` tag allows for executing any executable as a local OS process.
```

hidmic commented on articles/151_roslaunch_xml.md:
```diff
@@ -227,8 +239,8 @@ All substitutions are enclosed by `$(...)`.
-`$(find-package <pkg-name>)`
-: Substituted by the path to package installation directory in the local filesystem.
+`$(find-pkg <pkg-name>)`
+: Substituted by the install prefix path of the package.
```
@ivanpauno consider
```suggestion
: Substituted by the install prefix path of the given package.
```
instead.

hidmic commented on articles/151_roslaunch_xml.md:
```diff
@@ -238,6 +250,11 @@ All substitutions are enclosed by `$(...)`.
 
+`$(exec-in-package <exec-name> <package-name>)`
+: Substituted by the path to the executable in the local filesystem.
+  Lookups at the `lib` folder of the package.
```
@ivanpauno consider
```suggestion
  Executables are looked up in the `lib` directory of the package.
```

hidmic reviewed (approved):
LGTM for a couple tiny nits

wjwwood reviewed (approved):
@hidmic's suggestions make sense, but otherwise lgtm

ivanpauno committed:
Replace substitution syntax that were causing build problems

Signed-off-by: ivanpauno <ivanpauno@ekumenlabs.com>

ivanpauno committed:
Address per review comments

Signed-off-by: ivanpauno <ivanpauno@ekumenlabs.com>

ivanpauno commented:
@hidmic I've corrected #163's build problems in https://github.com/ros2/design/pull/247/commits/cb2130d6595f7f0f441985c56bc047021d0db3de (tested locally) and addressed your comments in https://github.com/ros2/design/pull/247/commits/d3e918010030114620b8b8272cae3e4e899046dd.

hidmic reviewed (approved):
LGTM!
